partiql-conformance-test-generator/src/lib.rs

```diff
@@ -1,174 +1,57 @@
-pub mod generator;
+mod generator;
+mod reader;
 mod schema;
-pub mod util;
+mod util;
+mod writer;
 
-use crate::schema::{
+use crate::generator::Generator;
+use crate::schema::spec::{
     Assertion, Assertions, Namespace, Namespaces, TestCase, TestCases, TestDocument,
 };
 use crate::util::StringExt;
-use ion_rs::value::owned::OwnedElement;
-use ion_rs::value::{Element, Sequence, Struct, SymbolToken};
-use ion_rs::IonType;
-use std::collections::HashSet;
-use std::ops::Add;
 
-// TODO: move these test data parsing functions to own file
-/// Converts a vector of Ion data into a `TestDocument`, which can be composed of `Namespace`s
-/// and `TestCase`s. `Namespace`s must be provided as IonLists while `TestCase`s must be provided
-/// as IonStructs. Other Ion types will result in a panic.
-///
-/// When encountering a duplicate namespace/test case, a namespace/test case will be added with
-/// '_0' suffixed to the end of the name.
-pub fn ion_data_to_test_document(all_ion_data: Vec<OwnedElement>) -> TestDocument {
-    let mut namespaces = Vec::new();
-    let mut test_cases = Vec::new();
+use std::path::Path;
 
-    let mut encountered_ns_names: HashSet<String> = HashSet::new();
-    let mut encountered_tc_names: HashSet<String> = HashSet::new();
+use crate::reader::read_schema;
 
-    for elem in all_ion_data {
-        match elem.ion_type() {
-            IonType::List => {
-                // namespace in document
-                let mut ns = test_namespace(&elem);
-                if encountered_ns_names.contains(&ns.name) {
-                    ns.name.push_str("_0")
-                }
-                encountered_ns_names.insert(ns.name.clone());
-                namespaces.push(ns)
-            }
-            IonType::Struct => {
-                // test case in document
-                let mut tc = test_case(&elem);
-                if encountered_tc_names.contains(&tc.test_name) {
-                    tc.test_name.push_str("_0")
-                }
-                encountered_tc_names.insert(tc.test_name.clone());
-                test_cases.push(tc)
-            }
-            _ => panic!("Document parsing requires an IonList or IonStruct"),
-        }
-    }
-    TestDocument {
-        namespaces,
-        test_cases,
-    }
+// TODO docs
+#[derive(Debug, Copy, Clone)]
+pub enum OverwriteStrategy {
+    Overwrite,
+    Backup,
```

Comment:
What will this enum be used for?

Reply:
This replaces the existing `TODO` for deleting the previously generated tests or backing them up before generating. It's still TODO, but modeled here rather than as a `//TODO` in build.rs.